src/components/scrollManager.js

```diff
@@ -0,0 +1,246 @@
+define([], function () {
+    'use strict';
+
+    const scrollIncrement = 40;
+    const scrollTime = 500;
+    const scrollExtra = Math.min(window.innerWidth, window.innerHeight)*0.05;
+
+    var scrollElement;
+    var scrollElementRect;
+    var scrollTarget = {};
+    var scrollTimer;
+    var oldActiveElement;
+
+    function clamp(min, value, max) {
+        return value < min ? min : value > max ? max : value;
+    }
+
+    function nearestScroll(v1, v2, min, max) {
+        var delta1 = v1 - min;
+        var delta2 = max - v2;
+        if (delta1 < 0 && delta1 < delta2) {
+            return delta1;
+        } else if (delta2 < 0) {
+            return -delta2;
+        }
+        return 0;
+    };
+
+    function trend(v, v1, v2) {
+        return (v1 < v && v2 < v1) || (v1 > v && v2 > v1) ? v2 : v1;
```

Comment:
I cannot grasp what this function does, please explain (or better expand it so it's clearer).
Or I might understand, but if I do then this function has a bug :) so an explanation is needed to understand whether I misunderstand or this has a bug.